packages/firestore/src/lite-api/aggregate.ts

```diff
@@ -15,40 +15,115 @@
  * limitations under the License.
  */
 
-import { DocumentData, Query, queryEqual } from './reference';
+import { invokeRunAggregationQueryRpc } from '../remote/datastore';
+import { cast } from '../util/input_validation';
+import { getDatastore } from './components';
+import { Firestore } from './database';
+import { Query, queryEqual } from './reference';
+import { LiteUserDataWriter } from './reference_impl';
 
-export class AggregateQuery<T = DocumentData> {
+/**
+ * A {@code AggregateQuery} computes some aggregation statistics from the result set of a base
+ * {@link Query}.
+ *
+ * <p><b>Subclassing Note</b>: Cloud Firestore classes are not meant to be subclassed except for use
+ * in test mocks. Subclassing is not supported in production code and new SDK releases may break
+ * code that does so.
+ */
+export class AggregateQuery {
   readonly type = 'AggregateQuery';
-  readonly query: Query<T>;
+  /**
+   * The query on which you called {@link countQuery} in order to get this
+   * `AggregateQuery`.
+   * Query type is set to unknown to avoid error caused by query type converter.
+   * might change it back to T after testing if the error do exist or not
+   */
+  readonly query: Query<unknown>;
 
   /** @hideconstructor */
-  constructor(query: Query<T>) {
+  constructor(query: Query<unknown>) {
```

Comment:
use Query<unknown> as proposed.
using<T> might conflict with custom convertor created by user, will test this down the road.